### DISCUSSION TITLE
Tags / Теги

### DISCUSSION THREAD

**Discussion by Duche3d** [PastVu/pastvu#20] [Ideas / Идеи и предложения]
Импотрирование тегов из старого сайта.
Ввод новых тегов.

**Comment by paul-k-pastvu** [comment]
Опять же, для этой группы задач/дискуссий [есть отдельная метка "Tags"](https://github.com/PastVu/pastvu/issues?q=is%3Aopen+label%3ATags).

Какие аспекты включает эта тема? На мой взгляд:
1. Концепция работы с тегами
2. Создание структуры тегов
3. Импорт сохраненных тегов со старого сайта в новую структуру (в т.ч. связка тегов с изображениями, с учетом замен)
4. Доработки интерфейса с учетом тегов (добавление панели тегов к фотографиям, подпискам, ленте комментариев, фильтру в галереях)
5. Права на добавление и редактирование тегов, отдельный флаг модератора тегов
6. Модерирование операций с тегами (предложения пользователей по присвоению/снятию/замене тега у фотографии, а также по добавлению новых тегов)
7. Логирование операций с самими тегами, а также со связкой тег+изображение
8. Администрирование тегов
9. Редактирование, объединение, удаление тегов (workfllow).

**Reply by paul-k-pastvu** [reply]
Часть вопросов по тегам обсуждаются в #11

**Comment by paul-k-pastvu** [comment]
От @Duche3d  по администрированию тегов (из #11):

----
ну я думаю нужно

1. иметь возможность вывести все одним списком все или поиском по сочетанию букв
2. иметь чекбоксы напротив тегов которые хочешь объединить и кнопку - объединить/редактировать, нажатием на которую и будет происходить корректная замена выбранных тегов
8000 это вначале трудно будет разобрать, но постепенно причешется
3. далее можно у тегов сделать статус "новый" - это теги который пользователи добавили, а не выбрали из списка уже имеющихся и соответственно отдельная кнопка для модераторов для просмотра и редактирования новых тегов, чтоб отслеживать состояние и недопускать разростание
4. и очень важное это поиск и уничтожение или объединение тегов которые назначены только одному фото - такие теги бессмысленны, поскольку инструмент тегов это в первую очередь инструмент создания подборок схожих фото, а не уникальная их идентификация (ну вообще в идеале в писке тегов нужно иметь информацию о количестве фото с соттветсвующим тегом)
5. Облако или список тегов, поиск по тегам - одним словом инструмент для просмотра сайта через теги. Сейчас это сделано довольно странно - открываешь какое-то фото, видишь у него тэг и тыкнув на него можешь увидеть другие фото с тем же тегом. Посмотреть какие теги есть или поискать по ним невозможно.

**Comment by anvaruk** [comment]
для начала бы поиск сделать

**Reply by paul-k-pastvu** [reply]
Какая связь? Я вот поиск даже в виде рамочной задачи не формулировал. Просто доработать индексацию, чтобы нормально искать Яндексом/Гуглом

**Reply by aeifn** [reply]
В поиске удобно искать определенный город или улицу.

**Reply by anvaruk** [reply]
тэги это же улучшение системы поиска, 
просто кликаешь на готовые слова, вместо набора

**Reply by paul-k-pastvu** [reply]
Вовсе нет. Это параллельный механизм навигации. 
В тегах может быть такое, чего нету ни в названии, ни в описании, ни в комментариях.
